x/interchainstaking/keeper/msg_server.go

```diff
@@ -317,3 +319,77 @@ func (k Keeper) EmitValsetRequery(ctx sdk.Context, connectionID string, chainID
 	)
 	return nil
 }
+
+// GovReopenChannel reopens an ICA channel.
+func (k msgServer) GovReopenChannel(goCtx context.Context, msg *types.MsgGovReopenChannel) (*types.MsgGovReopenChannelResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// validate the zone exists, and the format is valid (e.g. quickgaia-1.delegate)
+	parts := strings.Split(msg.PortId, ".")
```

Comment:
It is worth handling the submission of the 'icacontroller-' prefix, and strip it if it is there?

Reply:
i'll add validate basic i just backported what was there in v1.4.0